backend/src/services/auth/auth.repo.js

```diff
@@ -3,101 +3,146 @@
+ * Sessions are stored in the `sessions` table.
+ *
  * @param {string} sessionId - randomly generated UUID
- * @param {string} userId - id of the authenticated user
+ * @param {number} userId - id of the authenticated user (users.id)
  * @param {Date}   expiresAt - session expiration time
- * @returns {Promise<object>} session object
+ * @returns {Promise<object>} session record
  */
 async function createSession(sessionId, userId, expiresAt) {
-  const now = new Date();
-  const session = {
-    id: sessionId,
-    userId,
-    createdAt: now,
-    expiresAt,
-  };
-  sessions.set(sessionId, session);
+  const session = await prisma.sessions.create({
+    data: {
+      session_id: sessionId,
+      user_id: userId,
+      expires_at: expiresAt,
+    },
+  });
   return session;
 }
 
 /**
  * Look up the current user by session id.
+ * Returns null if the session is missing, expired, or the user no longer exists.
+ *
  * @param {string} sessionId
  * @returns {Promise<object|null>} user or null if not found/expired
  */
 async function getUserBySessionId(sessionId) {
-  const session = sessions.get(sessionId);
+  const session = await prisma.sessions.findUnique({
+    where: { session_id: sessionId },
+  });
+
   if (!session) {
     return null;
   }
 
   const now = new Date();
-  if (session.expiresAt && session.expiresAt <= now) {
+  if (session.expires_at && session.expires_at <= now) {
     // Session expired, clean it up.
-    sessions.delete(sessionId);
+    await prisma.sessions.delete({
+      where: { session_id: sessionId },
+    });
     return null;
   }
 
-  // Resolve the user associated with this session.
-  for (const user of users.values()) {
-    if (user.id === session.userId) {
-      return user;
-    }
+  // Resolve the user associated with this session from the database.
+  const user = await prisma.users.findUnique({
+    where: { id: session.user_id },
+  });
+
+  if (!user) {
+    // If the user row was deleted, also clean up the session.
+    await prisma.sessions.delete({
+      where: { session_id: sessionId },
+    });
+    return null;
   }
-  return null;
+
+  return user;
 }
 
 /**
- * Optional helper to invalidate a session, e.g. on logout.
+ * Invalidate a session, e.g. on logout.
+ *
  * @param {string} sessionId
  * @returns {Promise<void>}
  */
 async function deleteSession(sessionId) {
-  sessions.delete(sessionId);
+  await prisma.sessions.deleteMany({
+    where: { session_id: sessionId },
+  });
+}
+
+/**
+ * Update user profile fields and mark profile as complete.
+ *
+ * @param {number} userId
+ * @param {object} profileData - { first_name, last_name, pronouns }
+ * @returns {Promise<object>} updated user
+ */
+async function updateUserProfile(userId, profileData) {
+  const { first_name, last_name, pronouns } = profileData;
+
+  const user = await prisma.users.update({
+    where: { id: userId },
+    data: {
+      first_name,
+      last_name,
+      pronouns,
+      is_profile_complete: true,
```

Comment:
[Requires change]
We should check to ensure that `first_name` and `last_name` are non-null and non-empty and then accordingly set the value for `is_profile_complete`.

Reply:
Thanks for the updates! A small improvement would be to store the trimmed version of the names (or for that matter most of the user input strings) in the database. Maybe we could trim the input as soon as we destructure it from the request body and then we can simply use the extracted string.